### ISSUE TITLE
Games take a long time to launch when many fonts are installed

### ISSUE THREAD

**Issue by stuartpb** [ValveSoftware/Proton#4980]
This is more of a note for other people running into the same problem that I had - while it may be possible for Steam Play to solve / work around this, the slowdown/bottleneck identified by this issue is kind of an edge case (one that can cause [other issues](https://gitlab.com/inkscape/inkscape/-/issues/1773), and can usually be addressed by other means), so it's not really a pressing concern. However, I did spend several months scratching my head about why *my* system seemed to be the only one with this problem, so I'm going to detail it here.

I'm on Arch Linux, and I had the [ttf-google-fonts-git](https://aur.archlinux.org/packages/ttf-google-fonts-git/) AUR package installed, which adds the [thousands of fonts available on Google Fonts (in all variations)](https://fonts.google.com/) to the system. Wine processes [every font on the system](https://askubuntu.com/questions/1237065/wine-starts-slowly-due-to-font-abundance) every time the wine server starts up. This meant that, on my system, **every time** I tried to launch a game with Proton, I had to wait for **every font** to be processed - this ended up taking several (10-20) minutes for each game to launch.

Possible solutions to this problem:

- Reduce the number of fonts installed on your system. Since this was already causing issues with other programs on my computer, this was the route I took: I went ahead and replaced `ttf-google-fonts-git` on my system with just [`all-repository-fonts`](https://aur.archlinux.org/packages/all-repository-fonts/). Going forward, if I want to add a font from Google Fonts, I use [Font Manager](https://github.com/FontManager/font-manager) to do that on an as-needed basis.
- Steam could force a custom fontconfig with only a subset of "safe" fonts (ie. [those needed to run Steam itself](https://aur.archlinux.org/packages/steam-fonts/)) when launching games through Proton. (Most games don't need access to all the system's fonts, and this could be a good way to vendor in Windows-specific system fonts without needing them on the host.)
- Wine could maybe be refactored to not have this font enumeration at server startup, and/or Proton could share a Wine server across multiple games.

Again, reducing the number of fonts on the system returned me to acceptable (~1 minute) launch times.

**Comment by kisak-valve** [comment]
Upstream bug reports: https://bugs.winehq.org/show_bug.cgi?id=51367 https://bugs.winehq.org/show_bug.cgi?id=51371

**Comment by Naitrate** [comment]
Yep, can confirm that the amount of fonts that I had installed was slowing things down.

In my case, I selectively installed certain Nerdfonts, and completely removed Google's fonts (the ones that don't come with my distro) from my system.

**Comment by thingrammer** [comment]
I'm experiencing the same issue. Can you imagine the quantities of Chinese fonts? Take me several years to realise where the problem is.
